src/test/codegen/unchecked_shifts.rs

```diff
@@ -0,0 +1,64 @@
+// compile-flags: -O
+
+#![crate_type = "lib"]
+#![feature(unchecked_math)]
+
+// CHECK-LABEL: @unchecked_shl_unsigned_same
+#[no_mangle]
+pub unsafe fn unchecked_shl_unsigned_same(a: u32, b: u32) -> u32 {
+    // CHECK-NOT: and i32
+    // CHECK: shl i32 %a, %b
```

Comment:
annot: the LLVM instruction always gives poison by default for too-high shift amounts (<https://llvm.org/docs/LangRef.html#shl-instruction>), so no extra flags are expected in the IR.